components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -108,6 +108,38 @@ public String getIdOfUser(String userName) throws IdentityProviderException {
     }
 
     @Override
+    public String getEmailOfUser(String userId) throws IdentityProviderException {
+      Response userResponse = scimServiceStub.getUser(userId);
+        String userEmail;
+        if (userResponse == null) {
+            String errorMessage =
+                    "Error occurred while retrieving Id of user " + userId + ". Error : Response is null.";
+            log.error(errorMessage);
```

Comment:
We discussed to log only at service layer. So let's remove the logging here. Check for other places as well.

Reply:
yes, this should be removed. I'll remove it

Reply:
Shall we remove this?

Reply:
Instead of removing logs I suggest to use debug logs when doing debugging it will be useful

Reply:
this log removed now. but this method not used in service layer do we need to log it in here? @harsha89